[TFECO-12319] API DOCS For Registry Tagging and Visibility (#2682)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/api-docs/private-registry/modules/index.mdx

- If module namespace, name, and provider are specified, the specified provider for the given module is deleted along with all its versions.
- If only module namespace and name are specified, the entire module is deleted.

> For public modules, only the endpoint specifying the module namespace and name is valid. The other DELETE endpoints will 404.

Reviewer: ```suggestion
For public modules, you can only send `DELETE` requests when specifying the module namespace and name. Sending a `DELETE` request to a public module and specifying additional parameters returns a 404.
```

  --request DELETE \
  https://app.terraform.io/api/v2/organizations/my-organization/registry-modules/public/terraform-aws-modules/vpc/aws
```

## List All Tags for a Registry Module

> [](#list-tags-for-a-registry-module)

Reviewer: ```suggestion
```
BTW, I noticed too late that Bob added this reference-style markdown links everywhere after I tried to remove them from the response code tables 🤦   . They aren't hurting anything so I can take a task to remove them after we publish these docs so I can clean up my mess.

### Sample Request

[](#sample-request-2)

```bash
curl \
  --request GET \
  --header "Authorization: Bearer $TOKEN" \
  https://app.terraform.io/api/v2/registry-modules/mod-fZn7uHu99ZCpAKZJ/tag-bindings
```

### Sample Response

[](#sample-response-2)

```json
{
  "data": [
    {
      "id": "tb-yPdfdsfWQryyJ7X",
      "type": "tag-bindings",
      "attributes": {
        "key": "env",
        "value": "staging",
        "created-at": "2026-06-11T01:39:16.287Z"
      }
    },
    {
      "id": "tb-VebM7JLKPIynQ5Qx",
      "type": "tag-bindings",
      "attributes": {
        "key": "DEPT",
        "value": "ENGINEERING",
        "created-at": "2026-06-11T01:45:23.923Z"
      }
    },
    {
      "id": "tb-BN9cq80ofmJKL8973o",
      "type": "tag-bindings",
      "attributes": {
        "key": "billingID",
        "value": "1234",
        "created-at": "2026-06-09T23:20:06.883Z"
      }
    }
  ],
  "links": {
    "target": "/api/v2/organizations/my-organization/registry-modules/private/my-organization/my-module/aws"
  }
}
```

## Add or Update Tags to a Registry Module

[](#add-tags-to-a-registry-module)

Adds or updates tags for a registry module in your organization's registry.

> -> **Note:** Removing tags from a registry module is made via PATCH requests, where the request body will contain only the tag key and values that are currently needed.

Reviewer: ```suggestion

To remove tags from a registry module, send a `PATCH` request and omit the tags from the request body.
```

`PATCH /api/v2/organizations/:organization-name/registry-modules/:registry-name/:namespace/:module-name/:provider`

| Parameter | Description |
|---|---|
| `:organization_name` | The name of the organization that owns the registry module. |
| `:registry_name` | The name of the registry. For private modules, this should be `private` and for publicly curated modules, this should be `public`. |
| `:namespace` | The namespace of the module (typically your organization name for private modules). |
| `:module-name` | The name of the module. |
| `:provider` | The provider name. |

> 

Reviewer: ```suggestion

### Response codes

```

| Status | Response | Reason                                      |
|---|---|---------------------------------------------|
| [200][] | [JSON API document][] | Successfully added/updated the tag binding  |
| [400][] | [JSON API error object][] | Request body cannot be parsed|
| [401][] | [JSON API error object][] | User not authorized |
| [404][] | [JSON API error object][] | Registry module not found|
| [422][] | [JSON API error object][] | Unprocessable content - Tag key or value have a wrong format|

### Request Body

[](#request-body)

> This PATCH endpoint requires a JSON object with the following properties as a request payload.

Reviewer: ```suggestion
To send a `PATCH` request, you must include a JSON object with the following properties as a request payload.  
```

```

### Sample Response

[](#sample-response)

```json
{
  "data": {
    "id": "mod-e2EX43902FYe",
    "type": "registry-modules",
    "attributes": {
      "name": "iam",
      "namespace": "terraform-aws-modules",
      "provider": "aws",
      "version-statuses": [],
      "created-at": "2026-02-24T02:29:29.881Z",
      "updated-at": "2026-02-24T02:29:29.892Z",
      "registry-name": "public",
      "version-tags": [],
      "permissions": {
        "can-delete": true,
        "can-resync": true,
        "can-retry": true
      }
    },
    "relationships": {
      "organization": {
        "data": {
          "id": "organization-name",
          "type": "organizations"
        }
      },
      "tag-bindings": {
        "links": {
          "related": "/api/v2/registry-modules/mod-e2EX43902FYe/tag-bindings"
        }
      }
    },
    "links": {
      "self": "/api/v2/organizations/organization-name/registry-modules/public/terraform-aws-modules/iam/aws"
    }
  }
}
```
<!-- BEGIN: TFC:only -->
## Versions List for a Registry Module

[](#versions-list-for-a-registry-module)

> This endpoint returns all the published versions for a registry module with its tags' information.

Reviewer: ```suggestion
The following endpoint returns all published versions for a registry module including tags attached to each version.
```
